executor/grant.go

```diff
@@ -580,7 +580,7 @@ func (e *GrantExec) grantColumnLevel(priv *ast.PrivElem, user *ast.UserSpec, int
 		if err != nil {
 			return err
 		}
-		sqlexec.MustFormatSQL(sql, " WHERE User=%? AND Host=%? AND DB=%? AND Table_name=%? AND Column_name=%?", user.User.Username, user.User.Hostname, dbName, tbl.Meta().Name.O, col.Name.O)
+		sqlexec.MustFormatSQL(sql, " WHERE User=%? AND Host=%? AND DB=%? AND lower(Table_name)=%? AND Column_name=%?", user.User.Username, user.User.Hostname, dbName, strings.ToLower(tbl.Meta().Name.O), col.Name.O)
```

Comment:
does database name and column name also need this `lower`?

BTW, TiDB should behave like lower_case_table_names = 2. ref https://dev.mysql.com/doc/refman/5.6/en/identifier-case-sensitivity.html

Reply:
Done, please check! THKS